pkg/operator/ceph/csi/controller.go

```diff
@@ -332,8 +332,7 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
-	if !EnableCSIOperator() {
-
+	if !disableCSI && !EnableCSIOperator() {
```

Comment:
Is the goal of the setting `ROOK_CSI_DISABLE_DRIVER: true`  to completely skip all reconcile of the csi driver and csi operator? If so, seems like we should add a return statement above on [line 205](https://github.com/rook/rook/pull/14746/files#diff-6a89cde4909fd851e48da3acc4dd4628c2fdbd68c40226d0b6e789f8b36a62eeR205), so that we skip all reconciling in this method. If that is not the intention, then I'm not clear what is the intention of this flag since it's still reconciling some csi resources between lines 210-333.

Reply:
I had the same question today to @subhamkrai as I was under the same impression,

He mentioned `ROOK_CSI_DISABLE_DRIVER` should only responsible for not creating CSI driver from rook

And `enable_csi_operator` should be responsible for creating CSI operator.

So if `ROOK_CSI_DISABLE_DRIVER`:false and `enable_csi_operator`:true it should create the CSI operator resources 
But we can discuss this if needed more discussion and changes...

Reply:
@subhamkrai please add your thoughts ^^

Reply:
IMO, `ROOK_CSI_DISABLE_DRIVER` is the config that we should prioritize working correctly. This config has existed for a while, and I think we expect it to continue to be used after the CSI operator is GA'ed as well. This config has had the meaning "Rook should not manage CSI driver resources," and I think keeping that meaning will help others continue to use Rook as they have been without unexpected results.

IIUC, `ENABLE_CSI_OPERATOR` is something that I think is temporary while we are working to test and migrate to the new CSI operator setup. I think it's fine if `ENABLE_CSI_OPERATOR` doesn't do anything if `ROOK_CSI_DISABLE_DRIVER == true` since we don't expect to need the option for a long time.

I would suggest that `ROOK_CSI_DISABLE_DRIVER` behaves to disable all Rook management of CSI drivers, including the operator. I think that is also what Travis is expecting, based on his comment.

Reply:
Yes above should be the correct expectation 👍🏻

Reply:
We need to decide which flag we want to give precedence as let's say
1. User set `ROOK_CSI_DISABLE_DRIVER==true` and `ENABLE_CSI_OPERATOR == true`, if we place `return` higher it will skip deploying CSI Operator.

While working on this, I treated `ROOK_CSI_DISABLE_DRIVER` to enable/disable driver code and `ENABLE_CSI_OPERATOR` to enable/disable operator code.

We can also log an error when user sets ` `ROOK_CSI_DISABLE_DRIVER==true` and `ENABLE_CSI_OPERATOR == true`. But just to mention, we have to csi way of deployment `driver and operator` so to have it clear it clear for user I treated like this currently in the code.

Reply:
As per the huddle discussion,
We are keeping the current implementations,

Later when we remove the csi operator flag check, then we will bring back the stop the controller reconcile at start

Reply:
Per discussion, approving to go ahead with the current changes